backend/src/main/java/com/benine/backend/database/MySQLDatabase.java

```diff
@@ -232,12 +221,99 @@ public void addCamera(int id, String macAddress) {
   }
 
   @Override
+  public void checkCameras() throws SQLException {
+    ArrayList<Camera> cameras = ServerController.getInstance().getCameraController().getCameras();
+    ArrayList<String> macs = new ArrayList<String>();
+    ResultSet resultset = null;
+    Statement statement = null;
+    try {
+      statement = connection.createStatement();
+      String sql = "SELECT ID, MACAddress FROM camera";
+      resultset = statement.executeQuery(sql);
+      checkOldCameras(resultset, cameras, macs);
+      checkNewCameras(cameras, macs);
+    } catch (SQLException e) {
+      e.printStackTrace();
+      getLogger().log("Cameras could not be gotten from database.", LogEvent.Type.CRITICAL);
+    } catch (CameraConnectionException e) {
+      e.printStackTrace();
+    } finally {
+      if (statement != null) {
+        statement.close();
+        resultset.close();
```

Comment:
If 'statement == null` the `resultset` stream may not be closed, it this correct behavior?